Mission statement first draft (#11)
scverse/governance: in_progress/mission-statement.md

# Mission Statement

## Values

> * Efficiency – Work the largest amount of data possible given available resources. We are not just targeting the largest datasets, but making analysis of data as widely accessible as possible.  

Reviewer: Should this be the first point? Reads weird to me to have this as the first point :)

Basically, I think that the mission statement needs like 2 introductory sentences IMO.

Or move the values below the abstract.

Author: On positioning for values points – Not sure, had just drafted some points and wanted them in here for comment

I think we're known for efficiency, and to me it's a inclusivity issue (e.g. if you have unnecessarily high hardware requirements you are putting limitations on who can do science). I think it could use rephrasing/ ordering, but it has been a core value.

# Mission Statement

## Goals

> We want to make single cell data analysis as accesible as possible. This means

Reviewer 2: ```suggestion
We want to make single cell omics data analysis as accesible as possible. This means
```

Single cell can mean a lot on its own

* Interoperability -- we focus on developing open, standardized, and interchangeable file formats and keep our data openly accessible. 
* Diversity -- we believe in removing barriers to participation in research, including economic factors and institutional culture. Our tools will have the most impact if they are developed by and for a diverse community.

> * Transparency -- we do our work as openly as possible and welcome our community to join in the decision making process

Reviewer 3: ```suggestion
* Transparency -- we do our work as openly as possible and welcome our community to join the decision making process.
```

> * Simplicity – Invent as little as possible and work with standard pydata types. Our users already have to be experts in multiple-fields. Lets not raise the bar for usage higher.

Reviewer 2: ```suggestion
* Simplicity – Invent as little as possible and work with standard pydata types. We strive for easy-to-use interfaces that reduce mental overhead.
```

## Abstract

> Scanpy and accessory packages for single-cell data analysis in Python have seen a strong and steady increase in interest since its first publication.

Reviewer 2: ```suggestion
There has been increasing interest in using Python for single-cell omics data analysis, which has been catalyzed by Scanpy (single-cell analysis in Python) and its corresponding ecosystem.
```

> The ecosystem has outgrown a single package and modality, so we've formed a new organization, the **`scverse`**, to handle this.

Reviewer 2: ```suggestion
This ecosystem has outgrown a single package or modality and now routinely includes packages at the frontier of single-cell research and development. To handle the growing needs of this ecosystem, we've formed a new organization -- **`scverse`**.
```

> This ecosystem has outgrown a single package or modality and now routinely includes packages at the frontier of single-cell research and development. To handle the growing needs of this ecosystem, we've formed a new organization -- **`scverse`**.

Reviewer 2: ecosystem is used too many times here and it reads poorly

> The growing adoption of Python for single-cell omics data analysis has been catalyzed by [Scanpy](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1382-0) (single-cell analysis in Python) and its ecosystem.
> This ecosystem has outgrown a single package or modality and now routinely includes packages at the frontier of single-cell research and development. To handle the growing needs of this ecosystem, we've formed a new organization -- **`scverse`**.

Reviewer 3: ```suggestion
The growing adoption of Python for single-cell omics data analysis has been catalyzed by [Scanpy](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1382-0) and [AnnData](https://anndata.readthedocs.io/en/latest/). 
Around these two core packages, an "ecosystem" of packages has been contributed by various developers and institutions, extending single-cell analysis to different modalities and providing cutting-edge methods at the frontier of single-cell research. To further nurture the growth of this ecosystem, we've formed a new organization -- **`scverse`**.
```

> `Scverse` is a community-based ecosystem of tools with users and developers across many organizations.

Reviewer 3: ```suggestion

```

> For the ecosystem to be useful and grow, it’s essential that the core analytic tools are well made and continue to be maintained. Frequently, tools are made by small groups within individual labs, and can quickly fall into disrepair once a grad student leaves. To some extent, this is unavoidable and is the nature of academic code. For commonly used, important tools this needs to be avoided.

Reviewer 2: ```suggestion
To sustain the utility and growth of the ecosystem, it’s essential that the core analytic tools are robust and well-maintained. Accomplishing this entails consistently supporting and improving tools beyond what can be done in the conventional academic setting.
```

I think we can clean this part up,

Author: ```suggestion
To sustain the utility and growth of the ecosystem, it’s essential that the core analytic tools are robust and well-maintained. Accomplishing this entails consistently supporting and improving tools beyond what can be done in the conventional single-lab academic setting.
```

> To sustain the utility and growth of the ecosystem, it’s essential that the core analytic tools are robust and well-maintained. Accomplishing this entails consistently supporting and improving tools beyond what can be done in the conventional single-lab academic setting.

Reviewer 3: ```suggestion
To sustain the utility and growth of the ecosystem, it is essential that the core analytic tools are robust and well-maintained. Accomplishing this entails consistently supporting and improving tools beyond what can be done in the conventional single-lab academic setting.
```

> The scverse constitutes an environment of tools which is deliberately not unified under a single package, but instead a consortium of core analytic tools with shared maintanance resposibilities.

Reviewer 2: ```suggestion
Scverse constitutes an environment of tools which is deliberately not unified under a single package, but instead is a consortium of core analytic tools with shared maintenance responsibilities.
```

> Tools within this environment that are considered "core" for sharing data between tools or are required for fundamental analyses of single cell datatypes are placed under shared maintainance in the `scverse` github organization.

Reviewer 2: ```suggestion
A tool within this environment can be considered "core" if it is designed for sharing data between tools or is required for fundamental analyses of single cell datatypes. These tools are placed under shared maintenance in the `scverse` GitHub organization.
```

Corrected grammar, but this sentence is convoluted? Then what's not core?

Reviewer 2: is scverse just the core tools?

Author: ```suggestion
We define "core tools" as those which facilitate sharing data through common formats, or provide foundational support for single cell datatypes (e.g. modality specific IO, toolkits) and analyses. 
These tools are placed under shared maintenance in the `scverse` GitHub organization.
```

> Workflows in this environment are heterogeneous and this core can be supplemented with analysis specific accessory packages.

Reviewer 2: 1. what exactly is a workflow?
2. What is "this core" referring to
3.

> This document lays out our vision and processes around the project's community in a transparent and accessible way.

Reviewer 3: ```suggestion
This document lays out our vision and processes around the project's community.
```
(not necessary)

We are committed to keeping this community open to third-party contributions and usage and are committed to acknowledging contributions.

> This white paper may serve as an entry point for third parties into the structure of the scverse environment.

Reviewer 3: ```suggestion
This white paper may serve as an entry point for third parties into the structure of `scverse`.
```

## User engagement

> The ecosystem centred on `scverse` core packages consists of python packages which are documented on readthedocs and commonly accompanied by scientific publications.

Reviewer 2: ```suggestion
The ecosystem centered on `scverse` core packages consists of Python packages that are documented on readthedocs and commonly accompanied by scientific publications.
```

I don't understand what this is trying to say

Reviewer 2: `that are documented on readthedocs` -> `that follow best software engineering practices`

Reviewer 3: Can we even make such a statement about the "ecosystem"? We can guarantee best practice for "core" packages, but not necessarily the ecosystem?

Reviewer: Good point. Agree with @grst

> The ecosystem centred on `scverse` core packages consists of python packages which are documented on readthedocs and commonly accompanied by scientific publications.

Reviewer 3: ```suggestion
The ecosystem centered on `scverse` core packages consists of Python packages that follow best software engineering practices.
```

Its primary purpose is to allow users to generalise workflows to their own data analysis settings.
We promote user engagement via the following mechanisms:

> * Clear documentation of code and usage of tutorial notebooks.
> * Participation in development though issues and pull requests.
> * Community forums, such as our Discourse and Twitter account. scverse environment-centric workshops at which scverse workflows are taught and hackathons at which functionalities are contributed. We share event dates and resources within the environment to increase impact and to exploit synergies.

Reviewer 3: ```suggestion
* clear documentation of code and usage of tutorial notebooks,
* participation in development through issues and pull requests,
* community forums, such as our Discourse and Twitter account, 
* workshops at which we teach analysis workflows based on the `scverse` ecosystem, and
* hackathons, to lower the bar for code-contributions. 
```

> We share event dates and resources within the environment to increase impact and to exploit synergies.

"resources" and "environment" are quite unspecific, I guess we can as well delete that sentence.

## Developer engagement

The scverse heavily benefits from a variety of people contributing code via different channels. The channels used at the moment are:

* Pull requests on scverse core or ecosytem packages: These can be used to fix bugs that are experienced during a workflow or to contribute functionalities. 

> * Feedback/ bug reporting through issues and the community forum.

Reviewer 2: ```suggestion
* Feedback/bug reporting through issues and the community forum.
```

* Creating and sharing tutorial content, either through written documentation or in person workshops.
* Contributing and maintaining documentation, making it easier for people to use or understand available tools.

> * Participation in community forums, by helping new users or participating in discussions.

Author: ```suggestion
* Helping other users on community forums
* Participating in design discussions both for technical issues and what opportunities or challenges lie ahead for `scverse`.
```

> * Participation in community forums, by helping new users or participating in discussions.

Reviewer 3: ```suggestion
* Helping other users on community forums
* Participating in design discussions about both technical and strategic topics.
```

> Scverse heavily benefits from a variety of people contributing code via different channels. The channels used at the moment are:
> 
> * Pull requests on scverse core or ecosystem packages: These can be used to fix bugs that are experienced during a workflow or to contribute functionalities. 
> * Feedback/bug reporting through issues and the community forum.
> * Creating and sharing tutorial content, either through written documentation or in person workshops.
> * Contributing and maintaining documentation, making it easier for people to use or understand available tools.
> * Participation in community forums, by helping new users or participating in discussions.
> * Ecosystem packages: Development of tools and analyses which interact with core packages, work with AnnData or MuData objects, or further what can be done with single cell data in Python.

Reviewer 3: ```suggestion
Community contributions are instrumental to scverse. Apart from direct contributions of code through pull-requests to scverse core or ecosystem packages, we emphasize the importance of

    providing feedback and reporting bugs through the community forum and the packages' issue trackers,
    creating and sharing tutorial content, either through written documentation or in-person workshops,
    contributing and maintaining documentation, making it easier for people to use or understand available tools,
    participation in community forums, by helping new users or participating in discussions on what opportunities or challenges lie ahead for scverse, and
    developing ecosystem packages, that interact with scverse core packages and data-structures and further what can be done with single-cell data in Python.
```

Author: Ping @grst, I feel like there was a formatting issue here. Is this what you wanted it to look like?

Reviewer 3: huh, the bullet points vanished :scream: 
Fixed my suggestion now.

As a community, we try to incentivise code contributions through the following mechanisms:

* Pull requests: New workflows or methods can be contributed to the existing packages as extra functionalities in a pull request. These functionalities can be documented as being advised to be cited separately from the target repository if they are referenced in a scientific publication. Examples that were contributed to Scanpy are documented here: https://scanpy.readthedocs.io/en/docsearch/external/index.html. 
* Promotion of ecosystem packages: These packages can and are cited separately from Scanpy and can therefore further the developers academic careers.  Secondly, these packages are advertised on [scverse.org](), thus increasing the user base that is reached with the new package.

> * Core engagement in the community is acknowledged as contribution mentions on the website, which may certify the engagement in this community. Moreover, we can attribute contributions in our Twitter accounts to make this readily available for your career development.

Reviewer 2: only Twitter for career development?

Reviewer: Think that @falexwolf is maintaining the linkedin? Could we rebrand the Scanpy Linkedin to scverse?

Reviewer 3: * I would remove the career development statement, I feel it's irrelevant for the mission statement. 
 * This block feels partly redundant to the block above. Maybe we can first describe the code and ecosystem contributions, and then continue with the block above with something like 
 
 > Apart from direct contributions of code through pull-requests to scverse core or ecosystem packages, we emphasize the importance of

Reviewer 4: this conversation is outdated as the section has been removed, will resolve

Around this infrastructure an ecosystem of packages has been contributed by various developers and institutions — extending single-cell analysis to different modalities and addressing challenges at the cutting-edge of single-cell research.
To further nurture the growth of this ecosystem, we've formed a new organization -- **`scverse`**.

scverse is a consortium of tools with users and developers across the world. To sustain the utility and growth of the ecosystem, it’s essential the core analytic tools are robust and well-maintained. This entails consistently support and improvement tools beyond what's possible in the conventional single-lab academic setting.

Tools within `scverse` are deliberately not unified under a single package, and instead form a consortium of core analytic tools with shared maintenance responsibilities.
We define "core tools" as those which facilitate sharing data through common formats, or provide foundational support for single cell datatypes (e.g. modality specific IO, toolkits) and analyses. 
These tools are placed under shared maintenance in the `scverse` GitHub organization.

> Workflows in this environment are heterogeneous and this core can be supplemented with analysis specific accessory packages.
> This document lays out our vision and processes around the project's community.
> We are committed to keeping this community open to third-party contributions and are committed to acknowledging contributions.
> This white paper may serve as an entry point for third parties into the structure of the scverse environment.

Author: ```suggestion
We are committed to keeping this community open.
`scverse` core tools are expected to interface and be supplemented by other tools in the ecosystem.
However, development of core packages is not restricted.
Contributions to the core packages are welcomed and publicly acknowledged.

This document lays out our vision and processes around the project.
It may serve as an entry point for third parties into the structure of the scverse environment.
```

Author: Broader re-work of this section @grst @Zethson @adamgayoso @giovp

# Mission Statement

## Goals

> We want to make single cell omics data analysis as accessible as possible. This means

Reviewer: Note that our slogan is broader. Should this be more general, but also state here that we start with single-cell stuff?

> We want to make single cell omics data analysis as accessible as possible. This means

Reviewer 4: ```suggestion
We want to make for analysis tools for omics data in the life sciences as accessible as possible. This means
```

Reviewer 4: can I push this change @Zethson @grst ?

Reviewer: I honestly have very mixed feelings and will put this onto the agenda for tonight ok?

Reviewer 4: this was discussed, will resolve

* Diversity -- we believe in removing barriers to participation in research, including economic factors and institutional culture. Our tools will have the most impact if they are developed by and for a diverse community.
* Transparency -- we do our work as openly as possible and welcome our community to join the decision making process.
* Simplicity -- invent as little as possible and work with standard pydata types. We strive for easy-to-use interfaces that reduce mental overhead.

## Abstract

The growing adoption of Python for single-cell omics data analysis has been catalyzed by [Scanpy](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1382-0) and [AnnData](https://anndata.readthedocs.io/en/latest/). 
Around this infrastructure an ecosystem of packages has been contributed by various developers and institutions — extending single-cell analysis to different modalities and addressing challenges at the cutting-edge of single-cell research.
To further nurture the growth of this ecosystem, we've formed a new organization -- **`scverse`**.

> scverse is a consortium of tools with users and developers across the world. To sustain the utility and growth of the ecosystem, it’s essential the core analytic tools are robust and well-maintained. This entails consistent support and improvement tools beyond what's possible in the conventional single-lab academic setting.

Reviewer 4: ```suggestion
scverse is a consortium of tools with users and developers across the world. To sustain the utility and growth of the ecosystem, it’s essential the core analytic tools are robust and well-maintained. This entails consistent support and improvement of tools beyond what's possible in the conventional single-lab academic setting.
```

Tools within `scverse` are deliberately not unified under a single package, and instead form a consortium of core analytic tools with shared maintenance responsibilities.
We define "core tools" as those which facilitate sharing data through common formats, or provide foundational support for single cell datatypes (e.g. modality-specific IO, toolkits) and analyses. 
These tools are placed under shared maintenance in the `scverse` GitHub organization.

We are committed to keeping this community open.
`scverse` core tools are expected to interface with and be supplemented by other tools in the ecosystem.
However, development of core packages is not restricted – contributions to the core packages are welcomed and publicly acknowledged.

> This document lays out our vision and processes around the project.

Reviewer: That's what the title and the content so far indicates. Think we could even remove this sentence and the one after.

Reviewer 4: I could push this change, can we get another approval from @grst ?

Reviewer 4: I'll remove

It may serve as an entry point for third parties into the structure of the scverse environment.

## User engagement

> `scverse` provides a set of high quality core packages for performing analysis on single cell data.

Reviewer 4: ```suggestion
`scverse` provides a set of high quality infrastructure and analysis packages for single cell data.```

> These packages provide are well documented, tested, and provide broad functionality.

Reviewer 4: ```suggestion
These packages are well documented, tested, and provide broad functionality.
```

> All of these packages work with standardized data structures which use common python numeric types and have interchange-friendly on-disk formats.

Reviewer: ```suggestion
All of these packages work with standardized data structures which use common Python numeric types and have interchange-friendly on-disk formats.
```

> Beyond these standard packages, users have access to cutting-edge and alternative analysis methods through the large ecosystem of packages built around these core tools.

Reviewer 4: ```suggestion
Beyond standard packages, users have access to cutting-edge and alternative analysis methods through the large ecosystem of packages built around these core tools.
```

> Beyond the purely technical, we support our users through:

Reviewer 4: ```suggestion
Together with technical and development support, we also support our users through:
```

Reviewer 4: beyond used already above

* Diversity -- we believe in removing barriers to participation in research, including economic factors and institutional culture. Our tools will have the most impact if they are developed by and for a diverse community.
* Transparency -- we do our work as openly as possible and welcome our community to join the decision making process.
* Simplicity -- invent as little as possible and work with standard pydata types. We strive for easy-to-use interfaces that reduce mental overhead.

## Abstract

The growing adoption of Python for single-cell omics data analysis has been catalyzed by [Scanpy](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1382-0) and [AnnData](https://anndata.readthedocs.io/en/latest/). 
Around this infrastructure an ecosystem of packages has been contributed by various developers and institutions — extending single-cell analysis to different modalities and addressing challenges at the cutting-edge of single-cell research.
To further nurture the growth of this ecosystem, we've formed a new organization -- **`scverse`**.

> scverse is a consortium of tools with users and developers across the world. To sustain the utility and growth of the ecosystem, it’s essential the core analytic tools are robust and well-maintained. This entails consistently support and improvement tools beyond what's possible in the conventional single-lab academic setting.

Reviewer 3: ```suggestion
scverse is a consortium of tools with users and developers across the world. To sustain the utility and growth of the ecosystem, it’s essential the core analytic tools are robust and well-maintained. This entails consistent support and improvement tools beyond what's possible in the conventional single-lab academic setting.
```

Tools within `scverse` are deliberately not unified under a single package, and instead form a consortium of core analytic tools with shared maintenance responsibilities.

> We define "core tools" as those which facilitate sharing data through common formats, or provide foundational support for single cell datatypes (e.g. modality specific IO, toolkits) and analyses. 

Reviewer 3: ```suggestion
We define "core tools" as those which facilitate sharing data through common formats, or provide foundational support for single cell datatypes (e.g. modality-specific IO, toolkits) and analyses. 
```

These tools are placed under shared maintenance in the `scverse` GitHub organization.

We are committed to keeping this community open.

> `scverse` core tools are expected to interface and be supplemented by other tools in the ecosystem.

Reviewer 3: ```suggestion
`scverse` core tools are expected to interface with and be supplemented by other tools in the ecosystem.
```

However, development of core packages is not restricted – contributions to the core packages are welcomed and publicly acknowledged.

This document lays out our vision and processes around the project.
It may serve as an entry point for third parties into the structure of the scverse environment.

## User engagement

> 

Reviewer 3: ```suggestion
```

`scverse` provides a set of high quality core packages for performing analysis on single cell data.
These packages provide are well documented, tested, and provide broad functionality.

> All of these packages work with standardized data structures which use common python numeric types and have interchange friendly on-disk formats.

Reviewer 3: ```suggestion
All of these packages work with standardized data structures which use common python numeric types and have interchange-friendly on-disk formats.
```

* clear documentation of code and usage of tutorial notebooks,
* participation in development through issues and pull requests,
* community forums, such as our Discourse and Twitter account, 
* workshops at which we teach analysis workflows based on the `scverse` ecosystem, and

> * hackathons, to lower the bar for code-contributions. 

Author: ```suggestion
* hackathons, to focus and engage the developer community.
```

> * workshops at which we teach analysis workflows based on the `scverse` ecosystem

Reviewer 3: This block seems out of context?

Author: I've tried to add some, does that work for you

* providing feedback and reporting bugs through the community forum and the packages' issue trackers,
* creating and sharing tutorial content, either through written documentation or in-person workshops,
* contributing and maintaining documentation, making it easier for people to use or understand available tools,
* participation in community forums, by helping new users or participating in discussions on what opportunities or challenges lie ahead for scverse, and
* developing ecosystem packages, that interact with scverse core packages and data-structures and further what can be done with single-cell data in Python.

As a community, we try to incentivise code contributions through the following mechanisms:

* Pull requests: New workflows or methods can be contributed to the existing packages as extra functionalities in a pull request. These functionalities can be documented as being advised to be cited separately from the target repository if they are referenced in a scientific publication. Examples that were contributed to Scanpy are documented here: https://scanpy.readthedocs.io/en/docsearch/external/index.html. 

> * Promotion of ecosystem packages: These packages can and are cited separately from Scanpy and can therefore further the developers academic careers.  Secondly, these packages are advertised on [scverse.org](), thus increasing the user base that is reached with the new package.

Reviewer 3: ```suggestion
* Promotion of ecosystem packages: These packages can and are cited separately from Scanpy and can therefore further the developers academic careers.  Secondly, these packages are advertised on [scverse.org](https://scverse.org/projects/), thus increasing the user base that is reached with the new package.
```

> Community contributions are instrumental to scverse. Apart from direct contributions of code through pull-requests to scverse core or ecosystem packages, we emphasize the importance of
> 
> * providing feedback and reporting bugs through the community forum and the packages' issue trackers,
> * creating and sharing tutorial content, either through written documentation or in-person workshops,
> * contributing and maintaining documentation, making it easier for people to use or understand available tools,
> * participation in community forums, by helping new users or participating in discussions on what opportunities or challenges lie ahead for scverse, and
> * developing ecosystem packages, that interact with scverse core packages and data-structures and further what can be done with single-cell data in Python.
> 
> 
> As a community, we try to incentivise code contributions through the following mechanisms:
> 
> * Pull requests: New workflows or methods can be contributed to the existing packages as extra functionalities in a pull request. These functionalities can be documented as being advised to be cited separately from the target repository if they are referenced in a scientific publication. Examples that were contributed to Scanpy are documented here: https://scanpy.readthedocs.io/en/docsearch/external/index.html. 
> * Promotion of ecosystem packages: These packages can and are cited separately from Scanpy and can therefore further the developers academic careers.  Secondly, these packages are advertised on [scverse.org](https://scverse.org/projects/), thus increasing the user base that is reached with the new package.
> * Core engagement in the community is acknowledged as contribution mentions on the website, which may certify the engagement in this community. Moreover, we can attribute contributions in our Twitter accounts to make this readily available for your career development.

Author: Draft of a new direction for this section. 

@adamgayoso, could you suggest a description for how ecosystem packages can take advantage of the [discourse section](https://discourse.scverse.org/c/ecosytem/38)?

```suggestion
Apart from a collection of analysis tools, `scverse` aims to support the development of new approaches to analyze and work with single cell data.
We believe that progress in this field can't be silo'd to a few groups.

We support the development of third party tools by:

* Providing stable APIs to build on top of
* Providing standardized and well supported data structure to pass around data
* Making sure all contributions are recognized in our change logs and documentation
* Promoting 3rd party packages that rely on scverse via our websites and social media feeds.
* Establishing open communication channels for
  * Discussion and collaboration between developers
  * A common forum for user support and engagement
```
